Lib/test/test_clinic.py

```diff
@@ -338,6 +339,76 @@ def converter_init(self):
         )
         self.assertIn(msg, out)
 
+    @staticmethod
+    @contextlib.contextmanager
+    def _clinic_version(new_version):
+        """Helper for test_version_*() tests"""
+        _saved = clinic.version
+        clinic.version = new_version
+        try:
+            yield
+        finally:
+            clinic.version = _saved
+
+    def test_version_directive_lower(self):
+        with self._clinic_version('3'):
+            block = dedent("""
+                /*[clinic input]
+                version 2
```

Comment:
I think we can parametrize these tests.

Reply:
00a9557f5d4f86e2d1ac3f8e39d15c8d6502845e

Reply:
> 00a9557

Hmm, I know it's annoying when reviewers disagree, but I think I preferred it before 00a9557. I completely take @sobolevn's point that the tests were somewhat repetitive before 00a9557, and I agree that we should try to avoid repetitive tests where possible by using subtests. But I think tests that involve generating an expected output — rather than declaratively stating the expected output — are worse than repetitive tests. It becomes less clear what you're testing, and you risk bugs creeping into the test code.

I'm not convinced these tests in particular are particularly well suited for being parameterised.

Reply:
Actually, I guess I only object to `test_version_directive_insufficient_version()`. Generating the test _input_ is fine; just not keen on generating the expected _output_ of the test that we want to assert